Assets/Scripts/VoicelineManager.cs

```diff
@@ -65,11 +66,15 @@ public class VoicelineManager : MonoBehaviour
     
     public static AudioClip[] voicelines;
 
+    public static string[] PersistentVoicelineNames { get; private set; }
```

Comment:
High-level approach comment:
Don't separate persistent voicelines like this, it adds spaghetti. It'd be better to just track the stat change record for level progress (I believe I already set up an event for that, if not add one). Then call `TraverseOnLevelStart` (rename probably to `TraverseOnLevelProgression`). That way the system isn't fragmented. 

The rest of the changes seem fine, but Idk about the B3MIS Comm's Behavior, might need a refactor later.

We're gonna seriously want to setup some better persistence than using static variables. The stats system specifically will just be wiped every time you re-open the game, which sucks.

Reply:
Yeah, that would be a cleaner approach, but it's a lot of extra boilerplate for an event that already has a hook (the Start() method), and will never be triggered except for when that method is called.

And we won't ever need that stat for anything else because in the full version of the game there won't be levels at all. So I think the extra boilerplate isn't worth it.

Reply:
See this intrinsically makes it harder to do this correctly later cuz Start() does not map to level start necessarily when we might have multiple scenes at the same time. That's why LevelProgression is needed anyway. And you have to re-implement functionality to check what level started, which should be in a LevelProgression stat event.

It's not really that much "boilerplate", in fact, it'd be less code than you added here to avoid doing it that way.